modules/tinymce/src/core/main/ts/keyboard/EnterKey.ts

```diff
@@ -10,13 +10,14 @@ import { EditorEvent } from '../api/util/EventDispatcher';
 import VK from '../api/util/VK';
 import * as InsertNewLine from '../newline/InsertNewLine';
 import { endTypingLevelIgnoreLocks } from '../undo/TypingState';
+import { mockInputEvents } from './MockInputEvents';
 
 const handleEnterKeyEvent = (editor: Editor, event: EditorEvent<KeyboardEvent>) => {
   if (event.isDefaultPrevented()) {
     return;
   }
 
-  event.preventDefault();
+  mockInputEvents(editor, { inputType: 'insertParagraph', cancelable: false }, event)();
```

Comment:
This is incorrect in some cases, as this can insert either a new paragraph or a line break. So this will need to be `insertLineBreak` or `insertParagraph` depending on what `InsertNewLine.insert` does.